ios/Plugin/SecureStoragePluginImpl.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  SecureStoragePluginImpl.swift
+//  SecureStoragePluginImpl
+//
+//  Created by Laszlo Blum on 2023. 10. 27.
+//
+
+import Foundation
+import KeychainAccess
+
+public class SecureStoragePluginImpl: NSObject {
+    let keychain = Keychain(service: "cap_sec")
+    let keychainTmp = Keychain(service: "cap_sec_tmp")
+    
+    override init() {
+        super.init()
+    }
+    
+    public func set(key: String, value: String, accessibilityKey: String = "afterFirstUnlock") throws {
+        let accessibility = keychainAccessibilityModeLookup[accessibilityKey] ?? defaultAccessibility
+        let storedAccessibility = getAccessibility(key: key)
+        let needReAdd = storedAccessibility != nil && accessibility != storedAccessibility
+        
+        if needReAdd {
+            try copyToTmp(key: key, accessibility: accessibility)
+            try remove(key: key)
+        }
+        
+        try keychain
+            .accessibility(accessibility)
+            .set(value, key: key)
+        
+        if needReAdd {
+            try removeFromTmp(key: key)
+        }
+    }
+    
+    func getAccessibility(key: String) -> Accessibility?  {
+        do {
+            if let item = try keychain.get(key, handler: { $0 }), let rawValue = item["pdmn"] {
+                let accessibility = Accessibility(rawValue: rawValue as! String)
+                if accessibility != nil {
+                    return accessibility
+                }
+            }
+            return nil
+        } catch {
+            return nil
+        }
+    }
+    
+    func hasKey(key: String) -> Bool {
+        do {
+            let keys = keys()
+            return keys.contains(key)
+        }
+    }
+    
+    func copyToTmp(key: String, accessibility: Accessibility) throws {
```

Comment:
In case we have any issue between the delete and copy we lose data. This is why the temp storage for. In case of crashing, the plugin recovers by checking the values in the temp storage at next startup.